automation/test.sh

```diff
@@ -354,6 +355,26 @@ spec:
 EOF
 fi
 
+# label_filter value passed to Ginkgo --label-filter flag
+label_filter=""
```

Comment:
Why do we need this ? it is the state already isnt it ?

Reply:
Yes but implicitly, since its a very long script I had to go back and forward just to understand where $label_filster came from.

Reply:
I don't think we need it, we don't init variables in bash,
we didn't have it and all the targets in the block below it set the initial value

Reply:
The fact that there are no init variables in bash doesn't mean we shouldn't have them, compared to modern languages it lakes a lot more.

In this particular case, I think it helps understand the `add_to_label_filter` function, it mutates a global variable that is initialized nowhere.
Also,  the `$label_filter` is used before `add_to_label_filter` definition.

This script is very long and complex, I how we can improve it in a follow-up PR.

Reply:
This change is not mandatory to this PR, and we don't agree about it
so maybe please remove it and suggest a difference PR for the refactor ?
This way we won't block this PR

Reply:
I removed `$label_filter` for now.